SimPEG/Props.py

```diff
@@ -327,17 +327,27 @@ def _act_map_names(self):
         ])
 
     @property
-    def _needs_model(self):
+    def needs_model(self):
         """True if a model is necessary"""
         return len(self._act_map_names) > 0
 
+    @property
+    def _has_nested_models(self):
+        for k in self._props:
+            if (
+                    isinstance(self._props[k], properties.Instance) and
+                    issubclass(self._props[k].instance_class, HasModel)
+               ):
+                return True
+        return False
+
     @properties.validator('model')
-    def _check_model_length(self, change):
+    def _check_model_valid(self, change):
         """Checks the model length and necessity"""
         if change['value'] is properties.utils.undefined:
             return True
 
-        if len(self._act_map_names) == 0:
+        if not self.needs_model and not self._has_nested_models:
```

Comment:
Clarification question: nested models would by how you are working with parameters in Richards? Eg a "model" defined by multiple distinct parameters?